### PR TITLE
Improve txs notification

### PR DESCRIPTION
**Pull request by Lbqds**
Previously we used alarms to track transaction statuses, but the minimum trigger interval for alarms cannot be less than one minute. This caused some delay in sending notifications. This PR replaces alarms with timers to address this issue.

### CODE REVIEW

**Comment by Lbqds** [comment]
> Q: What is the downside of using setTimeout in backgroud.js vs. using browser alarms?

According to the [doc](https://developer.chrome.com/docs/extensions/migrating/to-service-workers/#convert-timers), the downside is that the `setTimeout/setInterval` can fail when the service worker is terminated.

The [doc](https://developer.chrome.com/docs/extensions/mv3/service_workers/service-worker-lifecycle/#idle-shutdown) provides the conditions for the service worker termination, and when the service worker restarts, the timer will also resume.

**Comment by polarker** [comment]
@Lbqds Given that timer does not work in the background when service worker is terminated, we should combine both approaches I guess.

1. A 1 minute refresh based on alarm, plus on-demand refresh based on timer.
2. The timer should not last more than 5 minutes for new transactions so as not to DoS the node/explorer backend when they are bugs.

WDYT?

**Comment by Lbqds** [comment]
> @Lbqds Given that timer does not work in the background when service worker is terminated, we should combine both approaches I guess.
> 
> 1. A 1 minute refresh based on alarm, plus on-demand refresh based on timer.
> 2. The timer should not last more than 5 minutes for new transactions so as not to DoS the node/explorer backend when they are bugs.
> 
> WDYT?

Sounds great 👍

File: packages/extension/src/background/alarms.ts [hunk 1]
```diff
@@ -15,6 +15,14 @@ export function setTransactionTrackerHistoryAlarm() {
   })
 }
 
+export const transactionTrackerUpdateAlarm = "core:transactionTracker:update"
+export function setTransactionTrackerUpdateAlarm() {
+  browser.alarms.create(transactionTrackerUpdateAlarm, {
+    delayInMinutes: 1,
+    periodInMinutes: 1  // fetch transaction updates of existing transactions every minute from onchain
```
Comment (h0ngcha0) [inline packages/extension/src/background/alarms.ts:22]: [Nit] Comment might not be accurate any more.

**Review by h0ngcha0 — APPROVED** [review]
LGTM :+1:

File: packages/extension/src/background/index.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { MessageType, messageStream } from "../shared/messages"
 import { getNetwork } from "../shared/network"
 import { migratePreAuthorizations } from "../shared/preAuthorizations"
 import { updateTokenList } from "../shared/token/storage"
+import { delay } from "../shared/utils/delay"
```
Comment (polarker) [inline packages/extension/src/background/index.ts:10]: This seems not needed any more

File: packages/extension/src/background/index.ts [hunk 3]
```diff
@@ -50,23 +50,7 @@ browser.alarms.onAlarm.addListener(async (alarm) => {
...
-    while (hasInFlightTransactions && runs < maxRetries) {
-      console.info(`~> waiting ${waitTimeInS}s for transaction updates`)
-      await delay(waitTimeInS * 1000)
-      console.info(
-        "~> fetching transaction updates as pending transactions were detected",
-      )
-      runs++
-      hasInFlightTransactions = await transactionTracker.update()
-    }
+    await transactionTracker.update()
```
Comment (polarker) [inline packages/extension/src/background/index.ts:53]: Btw, does this update both mainnet and testnet?

**Review by polarker — APPROVED** [review]
LGTM. Two minor comments.

File: packages/extension/src/background/index.ts (hunk 3 above)
Comment (Lbqds) [reply packages/extension/src/background/index.ts:53]: Yes, it updates each transaction status based on the network id.

Comment (polarker) [reply packages/extension/src/background/index.ts:53]: Perfect.

**Pull request merged**